e2e/app/component/edit-delete-annotations-modal.ts

```diff
@@ -0,0 +1,56 @@
+import {Page} from 'puppeteer';
+import {savePageToFile, takeScreenshot} from 'utils/save-file-utils'
+import {LinkText} from 'app/text-labels';
+import Modal from './modal';
+import {waitForText} from 'utils/waits-utils';
+import BaseElement from 'app/element/base-element';
+
+
+const modalTitle = 'Edit or Delete Review-Wide Annotation Fields';
+
+
+export default class EditDeleteAnnotationsModal extends Modal {
+
+  constructor(page: Page, xpath?: string) {
+    super(page, xpath);
+  }
+
+  async waitForLoad(): Promise<this> {
+    await super.waitForLoad();
+    try {
+      await waitForText(this.page, modalTitle, {xpath: this.getXpath()});
+    } catch (e) {
+      await savePageToFile(this.page);
+      await takeScreenshot(this.page);
+      throw new Error(`EditOrDeleteAnnotationFieldModal waitForLoad() encountered ${e}`);
+    }
+    return this;
+  }
+
+
+   /**
+    * @param {string} newName New name.
+    */
+  
+  // rename annotations field  name
+  async clickRenameAnnotationsName(newName?: string): Promise<void> {
+     await this.clickButton(LinkText.Rename);
+      
+     const selector = `${this.getXpath()}//*[contains(text(), "Edit")]/following::div[3]/input`;
```

Comment:
this xpath is prone to breaking. does `Textbox` class not working for this?

Reply:
Right! `Textbox `class not working here. In this case, when the rename button is clicked 
1. the cursor shows up in the textbox input field.
2. the existing text needs to be cleared.
3. then the newName is typed.